src/SignalR/server/Core/src/StreamTracker.cs

```diff
@@ -79,7 +87,7 @@ private static IStreamConverter BuildStream<T>()
         private interface IStreamConverter
         {
             Type GetItemType();
-            object GetReaderAsObject();
+            object GetReaderAsObject(bool isIAsyncEnumerable = false);
```

Comment:
Might be cleaner to pass the target type here and do the IsAsyncEnumerable check on the inside